### PR TITLE
Return error if the config file is invalid.

### PR DESCRIPTION
**Pull request by kislaykishore**
* Return error if config file is invalid
* Refactor the root command to make it testable.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -27,36 +27,34 @@ import (
...
+			return mountFn(configObj)
+		},
 	}
-}
-
-func init() {
 	cobra.OnInitialize(initConfig)
-	rootCmd.PersistentFlags().StringVar(&cfgFile, "config-file", "", "The path to the config file where all gcsfuse related config needs to be specified."+
-		"Refer to 'https://cloud.google.com/storage/docs/gcsfuse-cli#config-file' for possible configurations.")
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config-file", "", "Refer to 'https://cloud.google.com/storage/docs/gcsfuse-cli#config-file' for possible configurations.")
```
Comment (ashmeenkaur) [inline cmd/root.go:51]: Nit: is this intentional?

File: cmd/root.go [hunk 2]
```diff
@@ -72,12 +70,11 @@ func initConfig() {
...
 
-	err := viper.Unmarshal(&configObj, viper.DecodeHook(cfg.DecodeHook()), func(decoderConfig *mapstructure.DecoderConfig) {
+	cfgErr = viper.Unmarshal(&configObj, viper.DecodeHook(cfg.DecodeHook()), func(decoderConfig *mapstructure.DecoderConfig) {
 		// By default, viper supports mapstructure tags for unmarshalling. Override that to support yaml tag.
 		decoderConfig.TagName = "yaml"
+		// Reject the config file if any of the fields in the YAML don't map to the struct.
+		decoderConfig.ErrorUnused = true
 	},
 	)
-	if err != nil {
```
Comment (ashmeenkaur) [inline cmd/root.go:80]: Nit: Why not log fatal error in this case too?

File: main.go [hunk 3]
```diff
@@ -63,9 +65,21 @@ func main() {
 	// Make logging output better.
 	log.SetFlags(log.Ldate | log.Ltime | log.Lmicroseconds)
 	if strings.ToLower(os.Getenv("ENABLE_GCSFUSE_VIPER_CONFIG")) == "true" {
-		os.Args = convertToPosixArgs(os.Args)
-		cmd.Execute()
+		// TODO: implement the mount logic instead of simply returning nil.
+		rootCmd, err := cmd.NewRootCmd(func(config cfg.Config) error { return nil })
+		if err != nil {
+			exitOnError(err)
```
Comment (ashmeenkaur) [inline main.go:71]: We can use logger.Fatal here too?

Comment (kislaykishore) [reply main.go:71]: It seems logger.Fatal writes to stdout instead of stderr. I didn't want to change the failure behavior.

Comment (ashmeenkaur) [reply main.go:71]: When GCSFuse runs in foreground, logs go to stdout, when GCSFuse runs in background, logs go to sys logs and when log file is specified, logs go to log file. I think it is okay to write error log with logger.Fatal (to stdout) in this case as it has the benefit of capturing the log when GCSFuse is run in different mode. WDYT?

**Review by ashmeenkaur — APPROVED** [review]

**Pull request merged**